### PR TITLE
Grouping

### PR DESCRIPTION
**Pull request by PimPaardekooper2**
grouping can be turned off.

### CODE REVIEW

**Comment by Zeepaardje98** [comment]
Zet de png even in een gitignore

File: apocalypse_sim/data.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+"hello"
```
Comment (jtbuter) [inline apocalypse_sim/data.txt]: What is this?

File: apocalypse_sim/grid_map/map_layouts.py [hunk 2]
```diff
@@ -36,6 +36,11 @@ def __init__(self, map_id, model):
             maps = [self.simple_incubation]
         elif os.environ["mode"] == "4":
             maps = [self.doorway_map]
+        elif os.environ["mode"] == "5":
```
Comment (jtbuter) [inline apocalypse_sim/grid_map/map_layouts.py:39]: This mode can be removed as it was only used for reference when creating the poster

File: apocalypse_sim/grid_map/map_layouts.py [hunk 3]
```diff
@@ -433,15 +438,93 @@ def simple_incubation(self):
 
         return [city], [], []
 
+    def seperation_map(self):
```
Comment (jtbuter) [inline apocalypse_sim/grid_map/map_layouts.py:441]: This method can be removed.

File: apocalypse_sim/grid_map/map_layouts.py [hunk 4]
```diff
@@ -433,15 +438,93 @@ def simple_incubation(self):
...
+                      [self.model.grid.width, 0],
+                      [0, 0]],
+                     0, color="#ffffff")
+
+        human = Agents("human", [(1, 3), (4, 4), (4, 7),
+                                 (3, 5), (8, 0), (0, 8)])
+
+        return [city], [], [human]
+
+    def alignment_map(self):
```
Comment (jtbuter) [inline apocalypse_sim/grid_map/map_layouts.py:460]: This method can be removed.

File: apocalypse_sim/grid_map/map_layouts.py [hunk 5]
```diff
@@ -433,15 +438,93 @@ def simple_incubation(self):
...
+                      [0, 0]],
+                     0, color="#ffffff")
+
+        human = Agents("human", [(3, 1), (4, 4), (6, 5),
+                                 (2, 6), (6, 2), (5, 7),
+                                 (1, 3)])
+
+        return [city], [], [human]
+
+    def cohesion_map(self):
```
Comment (jtbuter) [inline apocalypse_sim/grid_map/map_layouts.py:480]: This method can be removed.

File: apocalypse_sim/grid_map/map_layouts.py [hunk 6]
```diff
@@ -433,15 +438,93 @@ def simple_incubation(self):
...
+                      [0, 0]],
+                     0, color="#ffffff")
+
+        human = Agents("human", [(3, 1), (4, 4), (1, 5),
+                                 (3, 6), (1, 3), (8, 5),
+                                 (8, 7), (7, 1), (1, 7), (0, 1)])
+
+        return [city], [], [human]
+
+    def avoid_zombie_search_brain(self):
```
Comment (jtbuter) [inline apocalypse_sim/grid_map/map_layouts.py:500]: This method can be removed

File: apocalypse_sim/grid_map/map_layouts.py [hunk 7]
```diff
@@ -433,15 +438,93 @@ def simple_incubation(self):
...
+        human = Agents("human", [(5,4), (2,3)])
+        zombie = Agents("zombie", [(2,6)])
+
+        return [city], [], [human, zombie]
 
 class Agents:
     """List of agents, same type different positions."""
 
-    def __init__(self, agent_type, positions):
+    def __init__(self, agent_type, positions, color=""):
```
Comment (jtbuter) [inline apocalypse_sim/grid_map/map_layouts.py:522]: Add description for color parameter

File: apocalypse_sim/model.py [hunk 8]
```diff
@@ -28,7 +28,7 @@ def __init__(self, height=50, width=50, density=0.1, infected_chance=0.05,
                  map_id=5, city_id=0, province="", total_agents=0,
                  human_kill_agent_chance=0.6, patient_zero=False,
                  door_width=5, seed=None, incubation_time=3,
-                 server=None):
+                 server=None, grouping=True, iteration=0):
         """Apocalypse object.
 
         Initializes the apocalypse object, makes the grid and put agents on
```
Comment (jtbuter) [inline apocalypse_sim/model.py:34]: Add grouping parameter description

File: apocalypse_sim/model_representation/mode.py [hunk 9]
```diff
@@ -24,6 +24,9 @@ def get_mode():
     elif os.environ["mode"] == "4":
         model_params, canvas_height, canvas_width, grid_height, grid_width = \
             doorway_mode()
+    elif os.environ["mode"] == "5":
```
Comment (jtbuter) [inline apocalypse_sim/model_representation/mode.py:27]: I think this mode is no longer needed

File: apocalypse_sim/model_representation/portrayals.py [hunk 10]
```diff
@@ -72,6 +73,10 @@ def human_portrayal(agent, portrayal):
         portrayal["Color"] = ["#80C904", "#4D7902"]
     else:
         portrayal["Color"] = ["#0000FF", "#9999FF"]
+
```
Comment (jtbuter) [inline apocalypse_sim/model_representation/portrayals.py]: This can be removed

File: apocalypse_sim/simulation-pim.py [hunk 11]
```diff
@@ -0,0 +1,108 @@
+from model import Apocalypse
```
Comment (jtbuter) [inline apocalypse_sim/simulation-pim.py]: I think this file can be removed in it's entirety, since it doesn't do multithreaded experiments.

**Review by jtbuter — CHANGES_REQUESTED** [review]

**Review by jtbuter — APPROVED** [review]

**Pull request merged**